exchanges/stream/websocket.go

```diff
@@ -486,40 +653,90 @@ func (w *Websocket) FlushChannels() error {
 	}
 
 	if w.features.Subscribe {
-		newsubs, err := w.GenerateSubs()
-		if err != nil {
-			return err
-		}
+		if w.GenerateSubs != nil {
+			newsubs, err := w.GenerateSubs()
+			if err != nil {
+				return err
+			}
 
-		subs, unsubs := w.GetChannelDifference(newsubs)
-		if w.features.Unsubscribe {
-			if len(unsubs) != 0 {
-				err := w.UnsubscribeChannels(unsubs)
+			subs, unsubs := w.GetChannelDifference(nil, newsubs)
+			if len(unsubs) != 0 && w.features.Unsubscribe {
+				err := w.UnsubscribeChannels(nil, unsubs)
 				if err != nil {
 					return err
 				}
 			}
+			if len(subs) < 1 {
+				return nil
+			}
+			return w.SubscribeToChannels(nil, subs)
 		}
-
-		if len(subs) < 1 {
-			return nil
+		for x := range w.connectionManager {
+			if w.connectionManager[x].Setup.GenerateSubscriptions == nil {
+				continue
+			}
+			newsubs, err := w.connectionManager[x].Setup.GenerateSubscriptions()
+			if err != nil {
+				if errors.Is(err, asset.ErrNotEnabled) {
+					continue
+				}
+				return err
+			}
+			subs, unsubs := w.GetChannelDifference(w.connectionManager[x].Connection, newsubs)
+			if len(unsubs) != 0 && w.features.Unsubscribe {
+				err = w.UnsubscribeChannels(w.connectionManager[x].Connection, unsubs)
+				if err != nil {
+					return err
+				}
+			}
+			if len(subs) != 0 {
+				err = w.SubscribeToChannels(w.connectionManager[x].Connection, subs)
+				if err != nil {
+					return err
+				}
+			}
 		}
-		return w.SubscribeToChannels(subs)
+		return nil
 	} else if w.features.FullPayloadSubscribe {
 		// FullPayloadSubscribe means that the endpoint requires all
 		// subscriptions to be sent via the websocket connection e.g. if you are
 		// subscribed to ticker and orderbook but require trades as well, you
 		// would need to send ticker, orderbook and trades channel subscription
 		// messages.
-		newsubs, err := w.GenerateSubs()
-		if err != nil {
-			return err
+
+		if w.GenerateSubs != nil {
+			newsubs, err := w.GenerateSubs()
+			if err != nil {
+				return err
+			}
+
+			if len(newsubs) != 0 {
+				// Purge subscription list as there will be conflicts
+				w.subscriptions.Clear()
+				return w.SubscribeToChannels(nil, newsubs)
+			}
+			return nil
 		}
 
-		if len(newsubs) != 0 {
-			// Purge subscription list as there will be conflicts
-			w.subscriptions.Clear()
-			return w.SubscribeToChannels(newsubs)
+		for x := range w.connectionManager {
+			if w.connectionManager[x].Setup.GenerateSubscriptions == nil {
+				continue
+			}
+			newsubs, err := w.connectionManager[x].Setup.GenerateSubscriptions()
+			if err != nil {
+				if errors.Is(err, asset.ErrNotEnabled) {
+					continue
+				}
+				return err
+			}
+			if len(newsubs) != 0 {
+				// Purge subscription list as there will be conflicts
+				w.connections[w.connectionManager[x].Connection].Subscriptions.Clear()
```

Comment:
We can see from how you've designed this, is that `w.connectionManager` is the source of truth, being that the map `w.connections` contains a pointer to the slice entry. You already have `w.connectionManager[x].Subscriptions` why not use that?